### PR TITLE
Namespaces for ActionCable channels

### PR DESCRIPTION
**Pull request by droptheplot**
In addition to #23185, this PR allows channels to be namespaced.

### CODE REVIEW

**Comment by rafaelfranca** [comment]
Could you write a test for it?

**Comment by droptheplot** [comment]
I'm not sure, but it works.

File: actioncable/test/server/configuration_test.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+require 'test_helper'
+require 'active_support/core_ext/array/grouping'
+
+class ActionCable::Server::ConfigurationTest < ActionCable::TestCase
+  setup do
+    @config = ActionCable::Server::Configuration.new(connection_class: ActionCable::Connection)
+  end
+
+  test "load namespaced channels" do
+    @config.channel_load_paths << File.join(FIXTURE_LOAD_PATH, 'channels')
```
Comment (palkan) [inline actioncable/test/server/configuration_test.rb]: I think, we can simply stub `channel_paths` method to test this functionality. We don't need _real_ files.

File: actioncable/lib/action_cable/server/configuration.rb [hunk 2]
```diff
@@ -28,7 +29,12 @@ def channel_paths
 
       def channel_class_names
         @channel_class_names ||= channel_paths.collect do |channel_path|
-          Pathname.new(channel_path).basename.to_s.split('.').first.camelize
+          dirname, basename = Pathname.new(channel_path).split
+
+          [
+            dirname.each_filename.to_a.split{ |f| f === 'channels' }.last,
```
Comment (droptheplot) [inline actioncable/lib/action_cable/server/configuration.rb]: > Why do we need "channels" part of path?

To get namespaces inside `channels` directory.
Yes, you're right, You should add `lib/channels` to `channel_load_paths`.

```ruby
ActionCable.server.config.channel_load_paths << Rails.root.join('lib/channels')
#=> [#<Pathname:/rails-test/app/channels>, #<Pathname:/rails-test/lib/channels>] 
```

**Comment by droptheplot** [comment]
Updated `channel_class_names ` method and tests, thanks to @palkan.

**Comment by eileencodes** [comment]
@droptheplot thanks for adding a test. Can you rebase and squash your commits so I can merge? Thanks!

**Comment by droptheplot** [comment]
@eileencodes because of changes in `server/configuration.rb` I had to make new commits. Now it looks good.

**Comment by matthewd** [comment]
Since a8df1bc345bf062e98bda64ef9638f65292c4b56, this should no longer be necessary.

**Pull request closed without merging**